### ISSUE TITLE
Sometimes traefik does not synchronize the data in redis

### ISSUE THREAD

**Issue by renl0314** [traefik/traefik#7531]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
I store my configuration in Redis.Sometimes when I modify redis data, traefik does not synchronize the modified data.Every time this happens, I need to restart the Traefik service.The configuration will not take effect until you restart。But that doesn't happen every time.

### What did you expect to see?

When I modify the data in redis, traefik synchronizes these configurations immediately.

### What did you see instead?

When this bug occurred, there was no log.I checked the data in the Traefik Dashboard and Redis, and they didn't match.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
v2.2.8
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Here is my Redis config stored in /etc/traefik/traefik.toml
```toml
[providers.redis]
    endpoints = ["traefik-redis:6379"]
    rootKey = "traefik"
    password = "1"
```

Docker-compose file:
```
version: '3'
services:
  traefik:
     image: traefik:v2.2.8
     restart: always
     container_name: traefik
     command: --api.insecure=true --providers.docker
     privileged: true
     ports:
       - "9080:80"
       - "9090:8080"
       - "443:443"
       - "4399:4399"
       - "4398:4398"
     volumes:
       - ./dynamic/:/dynamic/
       - ./static/:/etc/traefik/
       # So that Traefik can listen to the Docker events
       - /var/run/docker.sock:/var/run/docker.sock

   traefik-redis:
     image: redis:5.0.8
     restart: always
     container_name: traefik-redis
     privileged: true
     ports:
       - "6379:6379"
     volumes:
       - ./redis/config:/etc/config
       - ./redis/data:/data
     command: redis-server /etc/config/redis.conf
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
(paste your output here)
```

**Comment by leshik** [comment]
Maybe related to https://github.com/traefik/traefik/issues/6832#issuecomment-633678161. I'm seeing this as well, the issue is quite annoying.

**Comment by rtribotte** [comment]
Hello @renl0314,

Thanks for your interest in Traefik!
Did your restarted your redis server before it happens ? (like in #6832)

**Comment by leshik** [comment]
@rtribotte I'm seeing this as well. It happens even with no Redis restart. I suppose some short networking issues may cause this. The session becomes dead and that's it. When it happens, the Traefik IP address disappears from the Redis `CLIENT LIST`.

**Comment by rtribotte** [comment]
@leshik thanks for your feedbacks, it also looks like the same issue for me.

I'll close this issue as a duplicate in favor of #6832.
Thus, @renl0314, I encourage you to participate in the other issue, and also don't hesitate to ask for reopening this one, if it's a misunderstanding.

Thank you for your contribution.

**Comment by leshik** [comment]
@rtribotte I've seen this again today, and I think this might be a different thing than #6832 as in my case Redis wasn't restarted, and its IP address hasn't changed. What I'm seeing is likely the issue described [here](https://github.com/go-redis/redis/issues/938) and [here](https://github.com/go-redis/redis/issues/949). I wonder how this could be resolved, maybe it's possible to increase the number of pings or the timeout before the subscription is removed from Redis?

**Comment by leshik** [comment]
Looks like `valkeyrie` depends on the very old version of `go-redis`, the issue might be fixed already in the upstream.

**Comment by leshik** [comment]
@rtribotte I've built `valkeyrie` against `go-redis v.7.4.0`, have rebuilt `traefik` with it, but it didn't help, unfortunately. Less than 24h after `traefik` IP disappeared from Redis `CLIENT LIST`. No restart of any services has happened.

**Comment by renl0314** [comment]
> Hello @renl0314,
> 
> Thanks for your interest in Traefik!
> Did your restarted your redis server before it happens ? (like in #6832)

I am very sorry for replying you so late.I didn't restart my traefik service before it happens.I found that when I created a traefik service and a new redis service, I wrote thousands of pieces of data into redis at once.This is bound to happen.

**Comment by ldez** [comment]
This issue was closed in favor of #6832, could you continue the discussion inside #6832.
Thanks.

**Comment by renl0314** [comment]
> This issue was closed in favor of #6832, could you continue the discussion inside #6832.
> Thanks.

OK
